ui/user/src/lib/components/navbar/Tools.svelte

```diff
@@ -38,9 +44,24 @@
 
 	async function fetchThreadTools() {
 		if (!currentThreadID) return;
-		tools =
-			(await ChatService.listProjectThreadTools(project.assistantID, project.id, currentThreadID))
-				?.items ?? [];
+
+		// Get thread tools and tool references
+		const [threadTools, toolRefs] = await Promise.all([
+			ChatService.listProjectThreadTools(project.assistantID, project.id, currentThreadID),
+			ChatService.listAllTools()
+		]);
+
+		// Create a map of tool references by ID for faster lookup
+		const toolReferenceMap = new Map<string, ToolReference>();
+		for (const reference of toolRefs.items) {
+			toolReferenceMap.set(reference.id, reference);
+		}
+
+		// Enhance each tool with capability information
+		tools = (threadTools?.items || []).map((tool) => ({
+			...tool,
+			capability: toolReferenceMap.get(tool.id)?.metadata?.category === 'Capability'
+		}));
```

Comment:
alternatively, this field can be added to the assistant tools endpoint in the backend, but went with this for now.